model/src/pyrenew/process/differencedprocess.py

```diff
@@ -0,0 +1,228 @@
+            _integrate_one_step,
+            init=jnp.pad(
+                highest_order_diff_vals, (self.differencing_order, 0)
+            ),
+            xs=(
+                jnp.flip(jnp.arange(self.differencing_order)),
+                jnp.flip(init_diff_vals),
+            ),
+        )
+
+        return integrated
+
+    def sample(
+        self,
+        init_vals: ArrayLike,
+        n: int,
+        *args,
+        fundamental_process_init_vals: ArrayLike = None,
+        **kwargs,
+    ) -> tuple:
+        """
+        Sample from the process
+
+        Parameters
+        ----------
+        init_vals : ArrayLike
+            initial values for the differenced process,
+            passed as the init_diff_vals to
+            DifferencedProcess.integrate
+
+        n : int
+            Number of values to sample. Will sample n - 1
+            values from self.fundamental_process.
+
+        *args :
+           Additional positional arguments passed to
+           self.fundamental_process.sample()
+
+        fundamental_process_init_vals : ArrayLike
+           Initial values for the fundamental process.
+           Passed as the `init_vals` keyword argument
+           to self.fundamental_process.sample().
+
+        **kwargs : dict, optional
+            Keyword arguments passed to self.fundamental_process.sample()
+
+        Returns
+        -------
+        SampledValue
+            Whose value entry is a single array representing the
+            undifferenced timeseries
+        """
+        diffs, *_ = self.fundamental_process.sample(
+            *args, n=(n - 1), init_vals=fundamental_process_init_vals, **kwargs
+        )
+        return (
+            SampledValue(
+                value=self.integrate(init_vals, diffs.value),
+                t_start=self.t_start,
+                t_unit=self.t_unit,
+            ),
+        )
+
+    @staticmethod
+    def validate():
+        """
+        Validates input parameters, implementation pending.
+        """
+        return None
+
+    @staticmethod
+    def assert_valid_differencing_order(differencing_order: any):
+        """
+        To be valid, a differencing order must
+        be an integer and must be strictly positive.
+        This function raises a value error if its
+        argument is not a valid differencing order.
+
+        Parameter
+        ---------
+        differcing_order : any
+            Potential differencing order to validate.
+
+        Returns
+        -------
+        None or raises a ValueError
+        """
+        if not isinstance(differencing_order, int):
+            raise ValueError(
+                "differcing_order must be an integer."
+                f"got {type(differencing_order)}"
+            )
+        if not differencing_order >= 1:
+            raise ValueError(
+                "differcing_order must be an integer "
+                "greater than or equal to 1. Got "
+                f"{differencing_order}"
```

Comment:
Can combine these into a single `if` and with a common error message.

Reply:
Don't want to compare to `1` unless it's an `int`, to avoid an uninformative error:

```python
if isinstance("test", int) or "test" >= 1:
    print("this should not print")
```

```python
TypeError: '>=' not supported between instances of 'str' and 'int'
```

Reply:
Just do `if not (isinstance(differencing_order, int) and differencing_order > 1):`